src/app/shared/entity/external-machine-deployment.ts

```diff
@@ -127,4 +127,11 @@ export class ExternalMachineDeploymentSpecPatch {
 export class ExternalClusterMDPhase {
   state: ExternalClusterMDState;
   statusMessage?: string;
+  aks?: AKSState;
+}
+
+// this property is used only for AKS.
+export class AKSState {
```

Comment:
IMO, we can rename 

- AKSNodePoolStatusState
- AKSNodePoolStatus


We can't name to - AKSNodePoolState because we are using for ENUM for columns or maybe we can change that and rename to AKSNodePoolState to this?

wdyt?